common/services/user-locations.js

```diff
@@ -0,0 +1,68 @@
+const axios = require('axios')
+
+const { AUTH_PROVIDERS, NOMIS_ELITE2_API } = require('../../config')
+const { decodeAccessToken } = require('../lib/access-token')
+const referenceDataService = require('./reference-data')
+
+async function getUserLocations(token) {
+  var agencies
+
+  const tokenData = decodeAccessToken(token)
+
+  switch (tokenData.auth_source) {
+    case 'auth':
+      agencies = await _getAuthAgencies(token)
+      break
+    case 'nomis':
+      agencies = await _getNomisAgencies(token)
+      break
+  }
+
+  const locations = await Promise.all(
+    agencies.map(async agencyId => {
+      return referenceDataService.getLocations({ nomisAgencyId: agencyId })
+    })
+  )
+
+  return locations.map(results => {
+    return results[0].id
+  })
+}
+
+async function _getAuthAgencies(token) {
+  const tokenData = decodeAccessToken(token)
+
+  const response = await axios.get(
+    AUTH_PROVIDERS.hmpps.groups_url(tokenData.user_name),
+    {},
+    _authHeaders(token)
+  )
+
+  return response.data.map(group => {
+    return group.groupCode.replace(/^PECS_/, '')
+  })
+}
+
+async function _getNomisAgencies(token) {
```

Comment:
I wonder if it might be more readable to make these return a promise rather than using async/await.

This method could then become:

```js
function _getNomisAgencies(token) {
  return axios
    .get(NOMIS_ELITE2_API.user_caseloads_url, {
      headers: _getAuthHeaders(token),
    })
    .then(caseLoad => caseLoad.caseLoadId)
}
```

And I think `_authHeaders` should return just the headers object, not the whole config object:

```js
function _getAuthHeaders(token) {
  return {
    Authorization: `Bearer ${token}`,
  }
}
```

Reply:
`_getAuthAgencies` can also be structured in a similar way.

Reply:
Thanks, I forgot about `then`. I couldn't get the fancy `map` syntax above to work but it's certainly better now.